clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -248,29 +249,34 @@ private RecordsIterator(LogEntry entry) {
 
             long wrapperRecordTimestamp = entry.record().timestamp();
             this.logEntries = new ArrayDeque<>();
+            // If relative offset is used, we need to decompress the entire message first to compute
+            // the absolute offset. For simplicity and because it's a format that is on its way out, we
+            // do the same for message format version 0
             try {
                 while (true) {
                     try {
                         LogEntry logEntry = getNextEntryFromStream();
-                        Record recordWithTimestamp = new Record(
-                            logEntry.record().buffer(),
-                            wrapperRecordTimestamp,
-                            entry.record().timestampType()
-                        );
-                        logEntries.add(new LogEntry(logEntry.offset(), recordWithTimestamp));
+                        if (entry.record().magic() > Record.MAGIC_VALUE_V0) {
```

Comment:
@ijuma : Does this check need to actually happen? Please correct me if I'm wrong, but my understanding is that the magic # being > 0 indicates that the message is compressed and the only way this private constructor would be called would be because the message is compressed and the initial compression check done in the [makeNext()](https://github.com/wiyu/kafka/pull/1/files#diff-d2b756cbdd92a98d4a211552da8ef2c4R307)